libraries/bot-connector/src/main/java/com/microsoft/bot/connector/authentication/OAuthConfiguration.java

```diff
@@ -10,16 +10,22 @@
 public class OAuthConfiguration {
     private String scope;
     private String authority;
+    private Boolean validateAuthority;
 
     /**
      * Construct with authority and scope.
-     * 
+     *
      * @param withAuthority The auth authority.
      * @param withScope     The auth scope.
+     * @param withValidateAuthority Whether the Authority should be validated.
      */
-    public OAuthConfiguration(String withAuthority, String withScope) {
+    public OAuthConfiguration(String withAuthority, String withScope, Boolean withValidateAuthority) {
         this.authority = withAuthority;
         this.scope = withScope;
+        if (withValidateAuthority == null) {
+            withValidateAuthority = true;
+        }
+        this.validateAuthority = withValidateAuthority;
```

Comment:
strange to modify the incoming parameter.
Why not
`this.validateAuthority = withValidateAuthority != null? withValidateAuthority : true;`
or similar?

Reply:
I can do that, it's okay!